```{r echo = FALSE}
knitr::opts_chunk$set(out.width = "100%")
```

```{r echo=FALSE}
library(cow)
```

# (PART\*) Run a Tool With Galaxy {-}   

# Overview {#run-tool-with-galaxy-overview}

This chapter contains instructions for the SARS-CoV-2 sequencing data FastQC analysis with Galaxy on AnVIL activity.

Galaxy is a free computing platform that allows users to upload or "fetch" data from online repositories and analyze that data with supported tools using a [point and click user interface](https://en.wikipedia.org/wiki/Point_and_click) (UI) rather than with a [command-line interface](https://en.wikipedia.org/wiki/Command-line_interface). Galaxy on AnVIL means that the benefits of AnVIL (such as secure data sharing, access to controlled access data) are combined with the benefits of running a reproducible analysis with Galaxy. 

Like all compute on AnVIL, Galaxy on AnVIL will incur costs. 

## Learning Objectives

This activity will teach you how to use the AnVIL platform to:

1. Launch Galaxy on AnVIL
1. Import data into Galaxy on AnVIL
1. Examine sequence data files (`.fastq` files)
1. Find, open, and run a tool (FastQC)
1. Examine the quality control summary report (Webpage)
1. Export your results for future viewing
1. Show how to shut down Galaxy on AnVIL

# Getting Started {#run-tool-with-galaxy-getting-started}

## Account Login

If you don't have an AnVIL account, you will need to create one using the steps below in the AnVIL Account Creation Dropdown section.

<details><summary>Expand for AnVIL account creation instructions</summary>

:::: {.borrowed_chunk}
```{r, echo = FALSE, results='asis'}
cow::borrow_chapter(
  doc_path = "child/_child_student_create_account.Rmd",
  repo_name = "jhudsl/AnVIL_Template"
)
```
::::

</details>


<br/>Once you have an account, you can login and access Terra by going to [`anvil.terra.bio`](https://anvil.terra.bio/). Open Terra, and you should be prompted to sign in with your Google account.

## Workspace Set Up

Workspaces are the building blocks of projects in Terra. Inside a Workspace, you can run analyses, launch interactive tools like RStudio and Galaxy, store data, and share results. 

Cloning an existing Workspace allows you to copy existing documentation, code, and/or data into your own experimental space. The data for this activity is already available on AnVIL, specifically the [`demos-galaxy-fastqc` workspace](https://anvil.terra.bio/#workspaces/anvil-outreach/demos-galaxy-fastqc), so you'll want to clone the repository. 

When cloning, AnVIL makes a copy of notebooks and code for you to modify. Data however, is linked back to the original Workspace through Data Tables, which saves space!

::: {.notice}
The screenshots included below show a different workspace than what this tutorial will be using. For this tutorial, you want to clone the [`demos-galaxy-fastqc` workspace](https://anvil.terra.bio/#workspaces/anvil-outreach/demos-galaxy-fastqc) 

Be sure to name the cloned workspace a unique name in step 6. We recommend adding `_yourfirstname_yourlastname` to the workspace name.
:::

:::: {.borrowed_chunk}
```{r, echo = FALSE, results='asis'}
# Specify variables
AnVIL_module_settings <- list(
  workspace_name = "demos-galaxy-fastqc",
  workspace_link = "https://anvil.terra.bio/#workspaces/anvil-outreach/demos-galaxy-fastqc",
  billing_project = "sri-lanka-2025"
)

cow::borrow_chapter(
  doc_path = "child/_child_student_workspace_clone.Rmd",
  repo_name = "jhudsl/AnVIL_Template"
)
```
::::

:::: {.borrowed_chunk}
```{r, echo = FALSE, results='asis'}
cow::borrow_chapter(
  doc_path = "child/_child_workspace_clone.Rmd",
  repo_name = "jhudsl/AnVIL_Template"
)
```
::::

# Activity {#run-tool-with-galaxy-activity}

## Launching Galaxy on AnVIL

:::: {.borrowed_chunk}
```{r, echo = FALSE, results='asis'}
cow::borrow_chapter(
  doc_path = "child/_child_galaxy_start.Rmd",
  repo_name = "jhudsl/AnVIL_Template"
)
```
::::

### Navigating Galaxy on AnVIL

:::: {.borrowed_chunk}
```{r, echo=FALSE, results='asis'}
cow::borrow_chapter(
  doc_path = "child/_child_galaxy_navigate.Rmd",
  repo_name = "jhudsl/AnVIL_Template"
)
```
::::

## Importing Data into Galaxy on AnVIL

When we cloned our workspace, our cloned workspace linked to the original data in the workspace! We will upload three files from the AnVIL workspace into Galaxy, though we need only one `fastq` data sequence file for our activity. The others will be used if you want to continue with a related activity that performs alignment and variant discovery after quality control. These three files are (1) the forward and (2) the reverse reads for our sample, as well as, (3) the reference genome for SARS-CoV-2. There are two sets of reads for our sample because the scientists who collected it used paired-end sequencing. The sample files we are looking at end in `fastq` because they are raw data from the sequencer. The reference genome ends in `.fasta` because it has already been cleaned up by scientists. 

1. Click on “Upload Data” in the Tools pane.  

```{r, echo=FALSE, fig.alt='Screenshot of the Galaxy homepage. The Upload Data link has been highlighted.'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g3257f1d5e0e_0_29")
```

2. Click on “Choose remote files” at the bottom of the popup.

```{r, echo=FALSE, fig.alt='Screenshot of the Galaxy Data upload popup page and upload options.'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1112")
```

::: {.notice}
If you had files locally on your computer that you wanted to upload, you would use the "Choose local file" button.

Or if you had files you wanted to import from a data repository like Zenodo, you would use the "Paste/Fetch data" button.

We're using the "Choose remote files" button because we have data in our AnVIL workspace that we can import into the Galaxy on AnVIL instance.
:::


3. Double-click the Workspace folder.

```{r, echo=FALSE, fig.alt="Screenshot of the Galaxy Data upload popup pane, highlighting the AnVIL workspace where your data is linked"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1117")
```

4. Upload the sample sequence data files

  - Double-click “Tables/”

```{r, echo=FALSE, fig.alt="Screenshot of the Galaxy Data upload popup pane, highlighting the Tables/ folder where linked data will be in a cloned workspace"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1122")
```

  - Then double-click “sample/”.

```{r, echo=FALSE, fig.alt="Screenshot of the Galaxy Data upload popup pane, highlighting the sample/ folder where the data sequence .fastq files will be found"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1126")
```

  - Click the two sample `.fastq` file checkboxes to select them.

```{r, echo=FALSE, fig.alt="Screenshot of the Galaxy Data upload popup pane, selecting the two data sequence .fastq files"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1132")
```

  - These files will be highlighted in green when ready. Click “Ok”.

```{r, echo=FALSE, fig.alt="Screenshot of the Galaxy Data upload popup pane, highlighting the selected data sequence .fastq files and the Ok button."}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1136")
```

<details><summary>Expand for Steps 5- 6: Upload the reference genome</summary>

5. Repeat steps 2 and 3 from above.

  - Click on “Choose remote files” at the bottom of the popup.

```{r, echo=FALSE, fig.alt='Screenshot of the Galaxy Data upload popup page and upload options.'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1143")
```

  - Double-click the Workspace folder.

```{r, echo=FALSE, fig.alt="Screenshot of the Galaxy Data upload popup pane, highlighting the AnVIL workspace where your data is linked"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1117")
```

6. Upload the reference genome file

  - Again, double click "Tables/".

```{r, echo=FALSE, fig.alt="Screenshot of the Galaxy Data upload popup pane, highlighting the Tables/ folder where linked data will be in a cloned workspace"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1122")
```

  - This time, double click "reference/".

```{r, echo = FALSE, fig.alt="Screenshot of the Galaxy Data upload popup pane, highlighting the reference/ folder where the reference genome fasta file will be found."}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g3257f1d5e0e_0_56")
```

  - Click the `fasta` file.

```{r, echo=FALSE, fig.alt = "Screenshot of the Galaxy Data upload popup pane, highlighting the reference genome .fasta file"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1148")
```

  - This file will be highlighted green and click "Ok".

```{r, echo=FALSE, fig.alt = "Screenshot of the Galaxy Data upload popup pane, highlighting the selected reference genome .fasta file and the Ok button."}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1153")
```

</details>

<br/>
7. Click “Start”  

```{r, echo=FALSE, fig.alt='Screenshot of the Galaxy Data upload popup page. All three files are ready to be imported and the Start button is highlighted.'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1159")
```

8. Once complete, click “Close”.

```{r, echo=FALSE, fig.alt="Screenshot of the Galaxy Data upload popup pane. All trhee files are highlighted green and the Close button is highlighted." }
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1164")
```

9. Confirm that the files uploaded successfully by looking at the file names in the Galaxy History pane.

Note that the files will be highlighted in green in the Galaxy History pane once they are uploaded and available.

```{r, echo=FALSE, fig.alt="Screenshot of the Galaxy homepage. The successfully uploaded files are boxed in green color."}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1169")
```

## Examining `fastq` sequence data files

We will examine data in `fastq` format. This is the typical output from an Illumina Sequencer, but also the standard format output from most sequencers.

1. Use your mouse and click on the eye icon (![eye button image](resources/images/eye.png){#id .class width=25 height=20px}) of the first `fastq` file (**VA_sample_forward_reads.fastq**). 

```{r, echo=FALSE, fig.alt='Screenshot of the Galaxy homepage. Highlighting the eye icon for the forward reads .fastq file.'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g3257f1d5e0e_0_67")
```

2. After clicking the eye icon, in the Main screen you will see something like this:

```{r, echo=FALSE, fig.alt='Screnshot of a fastq file in the middle panel of Galaxy. The data includes DNA sequences but also includes many coded characters, making it hard to understand.'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1174")
```

<details><summary>Expand for FASTQ files explained</summary>

For more information on the [contents of a FASTQ file, consider this resource from Illumina](https://knowledge.illumina.com/software/general/software-general-reference_material-list/000002211).

::: {.reflection}
QUESTIONS:

1. How many lines in a .fastq file represent an individual read?

2. What does each line represent?

3. Why is the final line for each read (the quality score) important?
:::

::: {.dictionary}
**Breakout Box: Learn more about quality scores**

To save space, the sequencer records an [ASCII character](http://drive5.com/usearch/manual/quality_score.html) to represent scores 0-42. For example 10 corresponds to “+” and 40 corresponds to “I”. FastQC (a tool we'll be using next) knows how to translate this. This way of encoding the data is often called “Phred” scoring.

What does 0-42 represent? These numbers, when plugged into a formula, tell us the probability of an error for that base. This is the formula, where Q is our quality score (0-42) and P is the probability of an error:

Q = -10 log10(P)

Using this formula, we can calculate that a quality score of 40 means only 0.00010 probability of an error!
:::


</details>

## Finding and Using FastQC

FastQC is a tool which aims to provide simple quality control checks on raw sequence data coming from high throughput sequencing pipelines. It provides a set of analyses which you can use to get a quick impression of whether your data has any problems of which you should be aware before doing any further analysis.

1. Use the search tools bar in the upper left (within the tools pane in Galaxy)

```{r, echo=FALSE, fig.alt = "Screenshot of Galaxy on AnVIL with an arrow pointing to the tool search bar in the upper left corner. THe tools icon is also highlighted on the far left in case you have to first navigate there to see the search bar."}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g32e0e3cbaf5_0_2")
```

2. Type `fast` to search for FastQC, and select the tool in the list below.

```{r, echo=FALSE, fig.alt="Highlighting searching part of hte FastQC tool's name in the search bar and then selecting FastQC from the list below."}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1195")
```

3. This will open the tool menu in the middle pane. 

```{r, echo=FALSE, fig.alt = "Screenshot of Galaxy on AnVIL showing the FastQC tool menu has been opened in the middle pane."}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1200")
```

4. Switch the version of FastQC to `0.73+galaxy0`

  - Select the Versions icon (3 cubes).

```{r, echo=FALSE, fig.alt="Screenshot of Galaxy on AnVIL highlighting the versions icon in the gray banner at the top of the middle pane"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g32e28ad1f0a_0_7")
```

  - Select "Switch to 0.73+galaxy0" from the dropdown menu.

```{r, echo=FALSE, fig.alt="Screenshot of Galaxy on AnVIL highlighting which version of FastQC to switch to in the dropdown menu for tool versions."}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g3257f1d5e0e_0_0")
```

  - Confirm that the version now says "Selected 0.73+galaxy0".

```{r, echo=FALSE, fig.alt="Screenshot of Galaxy on AnVIL highlighting what the version dropdown menu should display."}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g3257f1d5e0e_0_4")
```

5. Confirm or select the correct input for FastQC (the forward reads fastq file).

```{r, echo=FALSE, fig.alt="Screenshot of Galaxy on AnVIL highlighting the input for the tool, specifically selecting the forward reads fastq file for the 'Raw read data from your current history'"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g32e28ad1f0a_0_2")
```

6. Run FastQC by clicking the blue "Run Tool" button.

```{r, echo=FALSE, fig.alt = "Screenshot of Galaxy on AnVIL highlighting the blue 'Run Tool' button in the top right corner of the middle pane."}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g32e28ad1f0a_0_19")
```

7. After submitting the job to run, the middle pane should have a message highlighted in green. 

```{r, echo=FALSE, fig.alt = "Screenshot of Galaxy on AnVIL showing how the middle pane and history pane look after submitting a job before it successfully runs."}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1204")
```

The history pane should also list what will be the output(s) from the tool. Note, before the job has finished running, these output(s) will be highlighted in gray. While running, the output(s) will be highlighted in an orange cream color. And once the tool runs successfully, the output(s) will be highlighted in green.

```{r, echo=FALSE, fig.alt = "Screenshot of Galaxy on AnVIL showing the outputs highlighted in green within the history pane."}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g3257f1d5e0e_0_127")
```

## Examining the FastQC quality control summary report

We will examine the FastQC output in webpage or `html` format. This form of the output provides graphs and a flag of "Passed", "Warn", or "Fail" for each subsection within the quality control analysis.

1. Use your mouse and click on the eye icon (![eye button image](resources/images/eye.png){#id .class width=25 height=20px}) of the FastQC Webpage output (**FastQC on data 1: Webpage**). 

```{r, echo=FALSE, fig.alt='Screenshot of the Galaxy homepage. Highlighting the eye icon for the forward reads .fastq file.'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g3257f1d5e0e_0_67")
```

2. This will open up a summary report for the sequencing file in the middle pane that you can scroll.

```{r, echo=FALSE, fig.alt = "Screenshot of an example summary report from FastQC"}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g325209e7a14_0_1216")
```

<details><summary>Expand for FastQC summary report explained</summary>

For more information on the [contents of the output quality control summary report from FastQC, consider this resource from Michigan State](https://rtsf.natsci.msu.edu/genomics/technical-documents/fastqc-tutorial-and-faq.aspx)

::: {.reflection}
QUESTIONS:

1. Explore “Basic Statistics”. How many total reads are there? Have any been flagged as poor quality? What is the sequence length?

2. Explore “Per base sequence quality”. Based on the Basic Statistics, is 28-40 a good or bad quality score? 

3. Is it okay to proceed based on the per base sequence quality?
:::

</details>

## Exporting your results

In case you want to view the results later, you can download the file. 

1.  Click on the name of the results you want to export/save, and it will expand the info shown for that file. Click the floppy disk/save icon.

```{r, echo=FALSE, fig.alt = "Screenshot of Galaxy on AnVIL highlighting the floppy disk/save icon for the results that you want to export."}
ottrpal::include_slide("https://docs.google.com/presentation/d/1zre8qeE9x56RSjusIF3XcuHoxqGf30nfdvjwCoEMd8o/edit#slide=id.g3257f1d5e0e_0_107")
```

This will download a zip (compressed) file with the results to your computer. Uncompress it to view it locally.

## Shutting down Galaxy on AnVIL

:::: {.borrowed_chunk}
```{r, echo = FALSE, results='asis'}
cow::borrow_chapter(
  doc_path = "child/_child_galaxy_delete.Rmd",
  repo_name = "jhudsl/AnVIL_Template"
)
```
::::

# Next steps {#run-tool-with-galaxy-next-steps}

This tutorial is just a first step in working with this data, specifically considering the quality scores of one set of the sequencing reads. Next steps include running FastQC on the reverse reads and downstream applications such as alignment and variant detection. Walk-throughs for these next steps can be found in [this Genomic Data Science Community Network Lab Exercise Book](https://jhudatascience.org/GDSCN_Book_SARS_Galaxy_on_AnVIL/student-activity-guide.html#alignment) or [this AnVIL Demos Recording](https://www.youtube.com/watch?v=_uT0IWL1wso).