<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>11 Exercises | AnVIL Demos</title>
  <meta name="description" content="Description about Course/Book." />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="11 Exercises | AnVIL Demos" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Description about Course/Book." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="11 Exercises | AnVIL Demos" />
  
  <meta name="twitter:description" content="Description about Course/Book." />
  



<meta name="date" content="2025-06-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="assets/AnVIL_style/anvil_favicon.ico" type="image/x-icon" />
<link rel="prev" href="single-cell-with-bioconductor-preparation.html"/>
<link rel="next" href="instructor-guide.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
  
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-L4S1S1ZKYM"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-L4S1S1ZKYM');
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
<link rel="stylesheet" href="assets/AnVIL_style/anvil.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://anvilproject.org/" target="_blank"><img src="assets/AnVIL_style/logo-anvil.png" style="width: 80%; padding-left: 15px; padding-top: 8px;"</a>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this Series</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#anvil-monthly-demos"><i class="fa fa-check"></i>AnVIL Monthly Demos</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#anvil-collection"><i class="fa fa-check"></i>AnVIL Collection</a></li>
</ul></li>
<li class="part"><span><b>What is AnVIL</b></span></li>
<li class="chapter" data-level="1" data-path="what-is-anvil-overview.html"><a href="what-is-anvil-overview.html"><i class="fa fa-check"></i><b>1</b> Overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="what-is-anvil-overview.html"><a href="what-is-anvil-overview.html#background"><i class="fa fa-check"></i><b>1.1</b> Background</a></li>
<li class="chapter" data-level="1.2" data-path="what-is-anvil-overview.html"><a href="what-is-anvil-overview.html#about-this-demo"><i class="fa fa-check"></i><b>1.2</b> About this Demo</a></li>
<li class="chapter" data-level="1.3" data-path="what-is-anvil-overview.html"><a href="what-is-anvil-overview.html#skills-level"><i class="fa fa-check"></i><b>1.3</b> Skills Level</a></li>
<li class="chapter" data-level="1.4" data-path="what-is-anvil-overview.html"><a href="what-is-anvil-overview.html#learning-objectives"><i class="fa fa-check"></i><b>1.4</b> Learning Objectives</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="what-is-anvil-preparation.html"><a href="what-is-anvil-preparation.html"><i class="fa fa-check"></i><b>2</b> Preparation</a>
<ul>
<li class="chapter" data-level="2.1" data-path="what-is-anvil-preparation.html"><a href="what-is-anvil-preparation.html#review-key-concepts"><i class="fa fa-check"></i><b>2.1</b> Review Key Concepts</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="what-is-anvil-exercises.html"><a href="what-is-anvil-exercises.html"><i class="fa fa-check"></i><b>3</b> Exercises</a>
<ul>
<li class="chapter" data-level="3.1" data-path="what-is-anvil-exercises.html"><a href="what-is-anvil-exercises.html#launch-terra"><i class="fa fa-check"></i><b>3.1</b> Launch Terra</a></li>
<li class="chapter" data-level="3.2" data-path="what-is-anvil-exercises.html"><a href="what-is-anvil-exercises.html#clone-hprc-workspace"><i class="fa fa-check"></i><b>3.2</b> Clone HPRC Workspace</a></li>
<li class="chapter" data-level="3.3" data-path="what-is-anvil-exercises.html"><a href="what-is-anvil-exercises.html#start-a-cloud-environment"><i class="fa fa-check"></i><b>3.3</b> Start a Cloud Environment</a></li>
<li class="chapter" data-level="3.4" data-path="what-is-anvil-exercises.html"><a href="what-is-anvil-exercises.html#find-tidbits"><i class="fa fa-check"></i><b>3.4</b> Find Tidbits</a></li>
<li class="chapter" data-level="3.5" data-path="what-is-anvil-exercises.html"><a href="what-is-anvil-exercises.html#enter-terminal"><i class="fa fa-check"></i><b>3.5</b> Enter Terminal</a></li>
<li class="chapter" data-level="3.6" data-path="what-is-anvil-exercises.html"><a href="what-is-anvil-exercises.html#shut-down"><i class="fa fa-check"></i><b>3.6</b> Shut Down</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="what-is-anvil-instructor-guide.html"><a href="what-is-anvil-instructor-guide.html"><i class="fa fa-check"></i><b>4</b> Instructor Guide</a>
<ul>
<li class="chapter" data-level="4.1" data-path="what-is-anvil-instructor-guide.html"><a href="what-is-anvil-instructor-guide.html#timeline"><i class="fa fa-check"></i><b>4.1</b> Timeline</a></li>
</ul></li>
<li class="part"><span><b>Scale with Workflows</b></span></li>
<li class="chapter" data-level="5" data-path="scale-with-workflows-overview.html"><a href="scale-with-workflows-overview.html"><i class="fa fa-check"></i><b>5</b> Overview</a>
<ul>
<li class="chapter" data-level="5.1" data-path="scale-with-workflows-overview.html"><a href="scale-with-workflows-overview.html#skills-level-1"><i class="fa fa-check"></i><b>5.1</b> Skills Level</a></li>
<li class="chapter" data-level="5.2" data-path="scale-with-workflows-overview.html"><a href="scale-with-workflows-overview.html#learning-objectives-1"><i class="fa fa-check"></i><b>5.2</b> Learning Objectives</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="scale-with-workflows-preparation.html"><a href="scale-with-workflows-preparation.html"><i class="fa fa-check"></i><b>6</b> Preparation</a>
<ul>
<li class="chapter" data-level="" data-path="scale-with-workflows-preparation.html"><a href="scale-with-workflows-preparation.html#quickstart"><i class="fa fa-check"></i>Quickstart</a></li>
<li class="chapter" data-level="6.1" data-path="scale-with-workflows-preparation.html"><a href="scale-with-workflows-preparation.html#review-key-concepts-1"><i class="fa fa-check"></i><b>6.1</b> Review Key Concepts</a></li>
<li class="chapter" data-level="6.2" data-path="scale-with-workflows-preparation.html"><a href="scale-with-workflows-preparation.html#create-anvil-account"><i class="fa fa-check"></i><b>6.2</b> Create AnVIL account</a></li>
<li class="chapter" data-level="6.3" data-path="scale-with-workflows-preparation.html"><a href="scale-with-workflows-preparation.html#clone-workspace"><i class="fa fa-check"></i><b>6.3</b> Clone Workspace</a></li>
<li class="chapter" data-level="6.4" data-path="scale-with-workflows-preparation.html"><a href="scale-with-workflows-preparation.html#start-cloud-environment"><i class="fa fa-check"></i><b>6.4</b> Start Cloud Environment</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html"><i class="fa fa-check"></i><b>7</b> Exercises</a>
<ul>
<li class="chapter" data-level="7.1" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html#explore-dataset-catalog"><i class="fa fa-check"></i><b>7.1</b> Explore Dataset Catalog</a></li>
<li class="chapter" data-level="7.2" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html#explore-hprc-workspace"><i class="fa fa-check"></i><b>7.2</b> Explore HPRC Workspace</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html#what-is-a-workspace"><i class="fa fa-check"></i><b>7.2.1</b> What is a Workspace?</a></li>
<li class="chapter" data-level="7.2.2" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html#dashboard"><i class="fa fa-check"></i><b>7.2.2</b> Dashboard</a></li>
<li class="chapter" data-level="7.2.3" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html#data"><i class="fa fa-check"></i><b>7.2.3</b> Data</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html#combine-data-workspace"><i class="fa fa-check"></i><b>7.3</b> Combine Data Workspace</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html#open-your-workspace"><i class="fa fa-check"></i><b>7.3.1</b> Open your Workspace</a></li>
<li class="chapter" data-level="7.3.2" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html#open-jupyter-notebook"><i class="fa fa-check"></i><b>7.3.2</b> Open Jupyter Notebook</a></li>
<li class="chapter" data-level="7.3.3" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html#load-packages"><i class="fa fa-check"></i><b>7.3.3</b> Load Packages</a></li>
<li class="chapter" data-level="7.3.4" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html#retrieve-original-file-locations"><i class="fa fa-check"></i><b>7.3.4</b> Retrieve original file locations</a></li>
<li class="chapter" data-level="7.3.5" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html#exported-combined-data-table"><i class="fa fa-check"></i><b>7.3.5</b> Exported combined Data Table</a></li>
<li class="chapter" data-level="7.3.6" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html#session-info"><i class="fa fa-check"></i><b>7.3.6</b> Session Info</a></li>
<li class="chapter" data-level="7.3.7" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html#view-your-new-data-table"><i class="fa fa-check"></i><b>7.3.7</b> View your new Data Table</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html#explore-dockstore-workflows"><i class="fa fa-check"></i><b>7.4</b> Explore Dockstore Workflows</a></li>
<li class="chapter" data-level="7.5" data-path="scale-with-workflows-exercises.html"><a href="scale-with-workflows-exercises.html#run-qc-analysis-pipeline"><i class="fa fa-check"></i><b>7.5</b> Run qc-analysis-pipeline</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="scale-with-workflows-instructor-guide.html"><a href="scale-with-workflows-instructor-guide.html"><i class="fa fa-check"></i><b>8</b> Instructor Guide</a>
<ul>
<li class="chapter" data-level="8.1" data-path="scale-with-workflows-instructor-guide.html"><a href="scale-with-workflows-instructor-guide.html#timeline-1"><i class="fa fa-check"></i><b>8.1</b> Timeline</a></li>
<li class="chapter" data-level="8.2" data-path="scale-with-workflows-instructor-guide.html"><a href="scale-with-workflows-instructor-guide.html#example"><i class="fa fa-check"></i><b>8.2</b> Example</a></li>
</ul></li>
<li class="part"><span><b>Single Cell with Bioconductor</b></span></li>
<li class="chapter" data-level="9" data-path="single-cell-with-bioconductor-overview.html"><a href="single-cell-with-bioconductor-overview.html"><i class="fa fa-check"></i><b>9</b> Overview</a>
<ul>
<li class="chapter" data-level="9.1" data-path="single-cell-with-bioconductor-overview.html"><a href="single-cell-with-bioconductor-overview.html#skills-level-2"><i class="fa fa-check"></i><b>9.1</b> Skills Level</a></li>
<li class="chapter" data-level="9.2" data-path="single-cell-with-bioconductor-overview.html"><a href="single-cell-with-bioconductor-overview.html#learning-objectives-2"><i class="fa fa-check"></i><b>9.2</b> Learning Objectives</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="single-cell-with-bioconductor-preparation.html"><a href="single-cell-with-bioconductor-preparation.html"><i class="fa fa-check"></i><b>10</b> Preparation</a>
<ul>
<li class="chapter" data-level="10.1" data-path="single-cell-with-bioconductor-preparation.html"><a href="single-cell-with-bioconductor-preparation.html#review-background"><i class="fa fa-check"></i><b>10.1</b> Review Background</a></li>
<li class="chapter" data-level="10.2" data-path="single-cell-with-bioconductor-preparation.html"><a href="single-cell-with-bioconductor-preparation.html#create-anvil-account-1"><i class="fa fa-check"></i><b>10.2</b> Create AnVIL account</a></li>
<li class="chapter" data-level="10.3" data-path="single-cell-with-bioconductor-preparation.html"><a href="single-cell-with-bioconductor-preparation.html#clone-workspace-1"><i class="fa fa-check"></i><b>10.3</b> Clone Workspace</a></li>
<li class="chapter" data-level="10.4" data-path="single-cell-with-bioconductor-preparation.html"><a href="single-cell-with-bioconductor-preparation.html#start-cloud-environment-1"><i class="fa fa-check"></i><b>10.4</b> Start Cloud Environment</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="single-cell-with-bioconductor-preparation.html"><a href="single-cell-with-bioconductor-preparation.html#video-overview"><i class="fa fa-check"></i><b>10.4.1</b> Video Overview</a></li>
<li class="chapter" data-level="10.4.2" data-path="single-cell-with-bioconductor-preparation.html"><a href="single-cell-with-bioconductor-preparation.html#objectives"><i class="fa fa-check"></i><b>10.4.2</b> Objectives</a></li>
<li class="chapter" data-level="10.4.3" data-path="single-cell-with-bioconductor-preparation.html"><a href="single-cell-with-bioconductor-preparation.html#slides"><i class="fa fa-check"></i><b>10.4.3</b> Slides</a></li>
<li class="chapter" data-level="10.4.4" data-path="single-cell-with-bioconductor-preparation.html"><a href="single-cell-with-bioconductor-preparation.html#launching-rstudio"><i class="fa fa-check"></i><b>10.4.4</b> Launching RStudio</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="single-cell-with-bioconductor-exercises.html"><a href="single-cell-with-bioconductor-exercises.html"><i class="fa fa-check"></i><b>11</b> Exercises</a>
<ul>
<li class="chapter" data-level="11.1" data-path="single-cell-with-bioconductor-exercises.html"><a href="single-cell-with-bioconductor-exercises.html#loading-the-data"><i class="fa fa-check"></i><b>11.1</b> Loading the Data</a></li>
<li class="chapter" data-level="11.2" data-path="single-cell-with-bioconductor-exercises.html"><a href="single-cell-with-bioconductor-exercises.html#installing-isee"><i class="fa fa-check"></i><b>11.2</b> Installing <code>iSEE</code></a></li>
<li class="chapter" data-level="11.3" data-path="single-cell-with-bioconductor-exercises.html"><a href="single-cell-with-bioconductor-exercises.html#panel-display"><i class="fa fa-check"></i><b>11.3</b> Panel Display</a></li>
<li class="chapter" data-level="11.4" data-path="single-cell-with-bioconductor-exercises.html"><a href="single-cell-with-bioconductor-exercises.html#visualize-cell-type-assignment"><i class="fa fa-check"></i><b>11.4</b> Visualize Cell Type Assignment</a></li>
<li class="chapter" data-level="11.5" data-path="single-cell-with-bioconductor-exercises.html"><a href="single-cell-with-bioconductor-exercises.html#visualize-expression-of-a-single-gene"><i class="fa fa-check"></i><b>11.5</b> Visualize Expression of a Single Gene</a></li>
<li class="chapter" data-level="11.6" data-path="single-cell-with-bioconductor-exercises.html"><a href="single-cell-with-bioconductor-exercises.html#get-session-info"><i class="fa fa-check"></i><b>11.6</b> Get Session Info</a></li>
<li class="chapter" data-level="11.7" data-path="single-cell-with-bioconductor-exercises.html"><a href="single-cell-with-bioconductor-exercises.html#shutting-down"><i class="fa fa-check"></i><b>11.7</b> Shutting Down</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="instructor-guide.html"><a href="instructor-guide.html"><i class="fa fa-check"></i><b>12</b> Instructor Guide</a></li>
<li class="part"><span><b>Run a Tool With Galaxy</b></span></li>
<li class="chapter" data-level="13" data-path="run-tool-with-galaxy-overview.html"><a href="run-tool-with-galaxy-overview.html"><i class="fa fa-check"></i><b>13</b> Overview</a>
<ul>
<li class="chapter" data-level="13.1" data-path="run-tool-with-galaxy-overview.html"><a href="run-tool-with-galaxy-overview.html#learning-objectives-3"><i class="fa fa-check"></i><b>13.1</b> Learning Objectives</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="run-tool-with-galaxy-getting-started.html"><a href="run-tool-with-galaxy-getting-started.html"><i class="fa fa-check"></i><b>14</b> Getting Started</a>
<ul>
<li class="chapter" data-level="14.1" data-path="run-tool-with-galaxy-getting-started.html"><a href="run-tool-with-galaxy-getting-started.html#account-login"><i class="fa fa-check"></i><b>14.1</b> Account Login</a></li>
<li class="chapter" data-level="14.2" data-path="run-tool-with-galaxy-getting-started.html"><a href="run-tool-with-galaxy-getting-started.html#workspace-set-up"><i class="fa fa-check"></i><b>14.2</b> Workspace Set Up</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="run-tool-with-galaxy-activity.html"><a href="run-tool-with-galaxy-activity.html"><i class="fa fa-check"></i><b>15</b> Activity</a>
<ul>
<li class="chapter" data-level="15.1" data-path="run-tool-with-galaxy-activity.html"><a href="run-tool-with-galaxy-activity.html#launching-galaxy-on-anvil"><i class="fa fa-check"></i><b>15.1</b> Launching Galaxy on AnVIL</a>
<ul>
<li class="chapter" data-level="15.1.1" data-path="run-tool-with-galaxy-activity.html"><a href="run-tool-with-galaxy-activity.html#navigating-galaxy-on-anvil"><i class="fa fa-check"></i><b>15.1.1</b> Navigating Galaxy on AnVIL</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="run-tool-with-galaxy-activity.html"><a href="run-tool-with-galaxy-activity.html#importing-data-into-galaxy-on-anvil"><i class="fa fa-check"></i><b>15.2</b> Importing Data into Galaxy on AnVIL</a></li>
<li class="chapter" data-level="15.3" data-path="run-tool-with-galaxy-activity.html"><a href="run-tool-with-galaxy-activity.html#examining-fastq-sequence-data-files"><i class="fa fa-check"></i><b>15.3</b> Examining <code>fastq</code> sequence data files</a></li>
<li class="chapter" data-level="15.4" data-path="run-tool-with-galaxy-activity.html"><a href="run-tool-with-galaxy-activity.html#finding-and-using-fastqc"><i class="fa fa-check"></i><b>15.4</b> Finding and Using FastQC</a></li>
<li class="chapter" data-level="15.5" data-path="run-tool-with-galaxy-activity.html"><a href="run-tool-with-galaxy-activity.html#examining-the-fastqc-quality-control-summary-report"><i class="fa fa-check"></i><b>15.5</b> Examining the FastQC quality control summary report</a></li>
<li class="chapter" data-level="15.6" data-path="run-tool-with-galaxy-activity.html"><a href="run-tool-with-galaxy-activity.html#exporting-your-results"><i class="fa fa-check"></i><b>15.6</b> Exporting your results</a></li>
<li class="chapter" data-level="15.7" data-path="run-tool-with-galaxy-activity.html"><a href="run-tool-with-galaxy-activity.html#shutting-down-galaxy-on-anvil"><i class="fa fa-check"></i><b>15.7</b> Shutting down Galaxy on AnVIL</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="run-tool-with-galaxy-next-steps.html"><a href="run-tool-with-galaxy-next-steps.html"><i class="fa fa-check"></i><b>16</b> Next steps</a></li>
<li class="part"><span><b>Human Genetic Variation in MAGE</b></span></li>
<li class="chapter" data-level="17" data-path="human-genetic-variation-in-mage-overview.html"><a href="human-genetic-variation-in-mage-overview.html"><i class="fa fa-check"></i><b>17</b> Overview</a>
<ul>
<li class="chapter" data-level="17.1" data-path="human-genetic-variation-in-mage-overview.html"><a href="human-genetic-variation-in-mage-overview.html#learning-objectives-4"><i class="fa fa-check"></i><b>17.1</b> Learning Objectives</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="human-genetic-variation-in-mage-preparation.html"><a href="human-genetic-variation-in-mage-preparation.html"><i class="fa fa-check"></i><b>18</b> Preparation</a>
<ul>
<li class="chapter" data-level="18.1" data-path="human-genetic-variation-in-mage-preparation.html"><a href="human-genetic-variation-in-mage-preparation.html#review-background-1"><i class="fa fa-check"></i><b>18.1</b> Review Background</a></li>
<li class="chapter" data-level="18.2" data-path="human-genetic-variation-in-mage-preparation.html"><a href="human-genetic-variation-in-mage-preparation.html#create-anvil-account-2"><i class="fa fa-check"></i><b>18.2</b> Create AnVIL Account</a></li>
<li class="chapter" data-level="18.3" data-path="human-genetic-variation-in-mage-preparation.html"><a href="human-genetic-variation-in-mage-preparation.html#clone-workspace-2"><i class="fa fa-check"></i><b>18.3</b> Clone Workspace</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html"><i class="fa fa-check"></i><b>19</b> Exercises</a>
<ul>
<li class="chapter" data-level="19.1" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#explore-mage-workspace"><i class="fa fa-check"></i><b>19.1</b> Explore MAGE Workspace</a>
<ul>
<li class="chapter" data-level="19.1.1" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#review-key-concepts-2"><i class="fa fa-check"></i><b>19.1.1</b> Review Key Concepts</a></li>
<li class="chapter" data-level="19.1.2" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#scavenger-hunt"><i class="fa fa-check"></i><b>19.1.2</b> Scavenger Hunt</a></li>
</ul></li>
<li class="chapter" data-level="19.2" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#analysis-with-jupyterterminal"><i class="fa fa-check"></i><b>19.2</b> Analysis with Jupyter/Terminal</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#launch-jupyter"><i class="fa fa-check"></i><b>19.2.1</b> Launch Jupyter</a></li>
<li class="chapter" data-level="19.2.2" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#review-key-concepts-3"><i class="fa fa-check"></i><b>19.2.2</b> Review Key Concepts</a></li>
<li class="chapter" data-level="19.2.3" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#enter-terminal-1"><i class="fa fa-check"></i><b>19.2.3</b> Enter Terminal</a></li>
<li class="chapter" data-level="19.2.4" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#run-jupyter-notebook"><i class="fa fa-check"></i><b>19.2.4</b> Run Jupyter Notebook</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#bioconductor-with-rstudio"><i class="fa fa-check"></i><b>19.3</b> Bioconductor with RStudio</a>
<ul>
<li class="chapter" data-level="19.3.1" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#launch-rstudio"><i class="fa fa-check"></i><b>19.3.1</b> Launch RStudio</a></li>
<li class="chapter" data-level="19.3.2" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#review-key-concepts-4"><i class="fa fa-check"></i><b>19.3.2</b> Review Key Concepts</a></li>
<li class="chapter" data-level="19.3.3" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#use-anvilgcp"><i class="fa fa-check"></i><b>19.3.3</b> Use AnVILGCP</a></li>
<li class="chapter" data-level="19.3.4" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#run-r-notebook"><i class="fa fa-check"></i><b>19.3.4</b> Run R Notebook</a></li>
</ul></li>
<li class="chapter" data-level="19.4" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#workflows-with-galaxy"><i class="fa fa-check"></i><b>19.4</b> Workflows with Galaxy</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#launch-galaxy"><i class="fa fa-check"></i><b>19.4.1</b> Launch Galaxy</a></li>
<li class="chapter" data-level="19.4.2" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#review-key-concepts-5"><i class="fa fa-check"></i><b>19.4.2</b> Review Key Concepts</a></li>
<li class="chapter" data-level="19.4.3" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#run-fastqc"><i class="fa fa-check"></i><b>19.4.3</b> Run FastQC</a></li>
<li class="chapter" data-level="19.4.4" data-path="human-genetic-variation-in-mage-exercises.html"><a href="human-genetic-variation-in-mage-exercises.html#install-tool"><i class="fa fa-check"></i><b>19.4.4</b> Install Tool</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="human-genetic-variation-in-mage-wrap-up.html"><a href="human-genetic-variation-in-mage-wrap-up.html"><i class="fa fa-check"></i><b>20</b> Wrap Up</a>
<ul>
<li class="chapter" data-level="20.1" data-path="human-genetic-variation-in-mage-wrap-up.html"><a href="human-genetic-variation-in-mage-wrap-up.html#shut-down-compute"><i class="fa fa-check"></i><b>20.1</b> Shut Down Compute</a></li>
<li class="chapter" data-level="20.2" data-path="human-genetic-variation-in-mage-wrap-up.html"><a href="human-genetic-variation-in-mage-wrap-up.html#learn-more"><i class="fa fa-check"></i><b>20.2</b> Learn More</a></li>
<li class="chapter" data-level="20.3" data-path="human-genetic-variation-in-mage-wrap-up.html"><a href="human-genetic-variation-in-mage-wrap-up.html#provide-feedback"><i class="fa fa-check"></i><b>20.3</b> Provide Feedback</a></li>
</ul></li>
<li class="divider"></li>
<p style="text-align:center;"> <a href="https://github.com/jhudsl/OTTR_Template" target="blank" > This content was published with</a> <a href="https://bookdown.org/" target="blank"> bookdown by: </a> </p>
<p style="text-align:center;"> <a href="https://hutchdatascience.org/"> The Fred Hutch Data Science Lab </a></p>
<a href="https://hutchdatascience.org/" target="_blank"><img src="https://raw.githubusercontent.com/jhudsl/OTTR_Template/main/style-sets/fhdasl/copy_to_assets/big-dasl-stacked.png" style="width: 80%; padding-left: 34px; padding-top: 8px;"</a>
<p style="text-align:center; font-size: 12px;"> <a href="https://github.com/rstudio4edu/rstudio4edu-book/"> Style adapted from: rstudio4edu-book </a> <a href ="https://creativecommons.org/licenses/by/2.0/"> (CC-BY 2.0) </a></p>
<p style="padding-left: 40px;"><div class="trapezoid" style = "padding-left: 40px;"><span><a href="https://docs.google.com/forms/d/e/1FAIpQLScrDVb_utm55pmb_SHx-RgELTEbCCWdLea0T3IzS0Oj00GE4w/viewform?usp=pp_url&entry.1565230805=AnVIL+Demos"> Click here to provide feedback</a> <img src="assets/itcr_arrow.png" style=" width: 10%" ></span></div></p>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">AnVIL Demos</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<head>
  <meta name="viewport" content="width=device-width,minimum-scale=1.0,maximum-scale=10.0,initial-scale=1.0">
  <!--script src="https://kit.fontawesome.com/6a26f47516.js"></script-->
  <!--<script src="assets/hideOutput.js"></script>-->
</head>
        


<div class="hero-image-container"> 
  <img class= "hero-image" src="assets/AnVIL_style/dasl_thin_main_image.png">
</div>
<div id="single-cell-with-bioconductor-exercises" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">11</span> Exercises<a href="single-cell-with-bioconductor-exercises.html#single-cell-with-bioconductor-exercises" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The following exercises will walk you through the process of using iSEE to explore single cell RNA-seq data. You will use data that have already been prepared and is saved in a SingleCellExperiment object. You will use the <code>pbmc3k</code> dataset from the <code>TENxPBMCData</code> package, which contains gene expression profiles for 2,700 single peripheral blood mononuclear cells. Information on how the data were preprocessed can be found <a href="https://isee.github.io/iSEEWorkshopEuroBioc2020/articles/dataset.html">here</a>.</p>
<p>These exercises were adapted from <a href="https://isee.github.io/iSEEWorkshopEuroBioc2020/">iSEEWorkshopEuroBioc2020/</a>.</p>
<div class="notice">
<p>To follow along with these exercises, you will need to complete the steps described in the <a href="scale-with-workflows-preparation.html#scale-with-workflows-preparation">Preparation</a> guide for this demo.</p>
</div>
<div id="loading-the-data" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Loading the Data<a href="single-cell-with-bioconductor-exercises.html#loading-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First, we’ll want to load the data that we’ll be using. To do that, we’ll need to load the <code>AnVIL</code> package created by the Bioconductor team for interfacing with files on AnVIL. Luckily, it’s already installed.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="single-cell-with-bioconductor-exercises.html#cb4-1" tabindex="-1"></a><span class="fu">library</span>(AnVIL)</span></code></pre></div>
<p>Next, we’ll use the <code>avfiles_restore()</code> function from the <code>AnVIL</code> package to actually bring the data into our environment’s persistent disk.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="single-cell-with-bioconductor-exercises.html#cb5-1" tabindex="-1"></a><span class="fu">avfiles_restore</span>( </span>
<span id="cb5-2"><a href="single-cell-with-bioconductor-exercises.html#cb5-2" tabindex="-1"></a>  <span class="at">source =</span> <span class="st">&quot;sc_bioconductor_data.RData&quot;</span>,</span>
<span id="cb5-3"><a href="single-cell-with-bioconductor-exercises.html#cb5-3" tabindex="-1"></a>  <span class="at">namespace =</span> <span class="st">&quot;anvil-outreach&quot;</span>,</span>
<span id="cb5-4"><a href="single-cell-with-bioconductor-exercises.html#cb5-4" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;demos-single-cell-bioconductor&quot;</span></span>
<span id="cb5-5"><a href="single-cell-with-bioconductor-exercises.html#cb5-5" tabindex="-1"></a>)</span></code></pre></div>
<p>In the code above, we are copying the file <code>sc_bioconductor_data.RData</code> from the Workspace <code>demos-single-cell-bioconductor</code>. It was created under the <code>anvil-outreach</code> Billing Project.</p>
</div>
<div id="installing-isee" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Installing <code>iSEE</code><a href="single-cell-with-bioconductor-exercises.html#installing-isee" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Once you’ve loaded the data, you’ll want to install and load the <code>iSEE</code> library. You can easily install it into your own personal RStudio environment using the pre-installed <code>BiocManager</code> commands.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="single-cell-with-bioconductor-exercises.html#cb6-1" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&#39;iSEE&#39;</span>)</span>
<span id="cb6-2"><a href="single-cell-with-bioconductor-exercises.html#cb6-2" tabindex="-1"></a><span class="fu">library</span>(iSEE)</span></code></pre></div>
</div>
<div id="panel-display" class="section level2 hasAnchor" number="11.3">
<h2><span class="header-section-number">11.3</span> Panel Display<a href="single-cell-with-bioconductor-exercises.html#panel-display" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First we will take a look at the interactive plots that iSEE can display.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g24f58a64a6b_0_53.png" width="100%" /></p>
</div>
<div id="visualize-cell-type-assignment" class="section level2 hasAnchor" number="11.4">
<h2><span class="header-section-number">11.4</span> Visualize Cell Type Assignment<a href="single-cell-with-bioconductor-exercises.html#visualize-cell-type-assignment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next, let’s focus specifically on visualizing cell type assignment by cluster membership. The goal is to identify the predominant cell type in each cluster. We can do this by plotting the column data in a <code>ColumnDataPlot</code>.</p>
<p>First, select the panel organization button and select “Organize panels”.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g254b1a8b951_0_12.png" width="100%" /></p>
<p>Remove all plots except for the column data plot. This will make things easier to view. Change the width to <code>12</code>.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g24f58a64a6b_0_114.png" width="100%" /></p>
<p>You should now see a large scatter plot. Select “Data parameters” underneath the plot.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g254b1a8b951_0_100.png" width="100%" /></p>
<p>First, select “labels_fine” under “Column of interest (Y-axis)”. Directly below, select the “Column data” button for “X-axis”. Once the dropdown menu appears for “Column of interest (X-axis)”, select “Cluster”.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g24f58a64a6b_0_124.png" width="100%" /></p>
<p>Since both cell annotations and the cluster are categorical, <code>iSEE</code> will generate a visual representation of a matrix called a “Hinton plot”.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g24f58a64a6b_0_129.png" width="100%" /></p>
<div class="notice">
<p>Now we know that cluster 4 contains almost all the cells that were annotated as classical monocytes. On the other hand, T cells can be found in multiple clusters.</p>
</div>
<p>We can also save the R code used to create our iSEE plots. This helps make our work reproducible!</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g24f58a64a6b_0_134.png" width="100%" /></p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g24f58a64a6b_0_139.png" width="100%" /></p>
</div>
<div id="visualize-expression-of-a-single-gene" class="section level2 hasAnchor" number="11.5">
<h2><span class="header-section-number">11.5</span> Visualize Expression of a Single Gene<a href="single-cell-with-bioconductor-exercises.html#visualize-expression-of-a-single-gene" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now let’s take a look at the expression data of a single gene across all the clusters. We can use the “Feature assay plot” panel to plot the distribution of the logcount values for a particular gene.</p>
<p>Click on the “Organize Panels” icon in the top right corner. Remove the “column plot” and choose “feature assay plot”. Change the width of the plot to 12.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g254b1a8b951_0_33.png" width="100%" /></p>
<p>You should now see a rather underwhelming bar plot. We still need to change the data parameters, so click on the “Data parameters” box.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g254b1a8b951_0_130.png" width="100%" /></p>
<p>Next, change the “Y-axis feature” to “LYZ”. This is the gene whose expression we’ll be examining. Change the feature selection box to “logcounts.”</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g254b1a8b951_0_60.png" width="100%" /></p>
<div class="notice">
<p>The <strong>LYZ</strong> gene encodes an enzyme called <strong>lysozyme</strong>, which plays a crucial role in the immune system’s defense against bacterial infections. The primary function of lysozyme is to break down bacterial cell walls.</p>
<p>The highest levels of LYZ gene expression are typically observed in tissues with direct contact with the external environment, such as the epithelial cells of the respiratory tract, gastrointestinal tract, and genitourinary system. These tissues are often exposed to potential pathogens, and the expression of lysozyme helps provide an additional line of defense against bacterial invasion.</p>
</div>
<p>Next, click the “column data” button under the “X-axis” header. Finally, choose “Cluster” from the drop down menu of “X-axis column data.”</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g254b1a8b951_0_66.png" width="100%" /></p>
<p>Now we have a much more exciting violin plot of LYZ gene expression levels across the 14 clusters in our dataset. LYZ is expressed more in clusters 4, 8, 9, and 13. We might be interested in also displaying cell type information in this plot, which we can do using the Visual Parameters options. Click the “Visual parameters” box.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g254b1a8b951_0_77.png" width="100%" /></p>
<p>Make sure “Color” is checked in the first row. Choose “Column Data” under the “Color by” options and change the drop down menu to “labels_main”. We could also choose to color the data by “labels_fine”.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g254b1a8b951_0_46.png" width="100%" /></p>
<p>We should see the dots in our violin plot colored by cell type annotation. Three of the clusters which have higher LYZ expression contain large numbers of cells identified as monocytes. Since LYZ codes for a human lysosome protein and is often used as a marker gene for monocytes, this makes a lot of sense.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g254b1a8b951_0_50.png" width="100%" /></p>
<div class="notice">
<p><strong>Monocytes</strong> are a subset of white blood cells that play a pivotal role in our immune defense against infections. Upon encountering an infection or inflammation, they migrate from the bloodstream to the affected tissues, where they differentiate into specialized cells that engulf and eliminate pathogens.</p>
</div>
<p>Let’s download this plot. Click on the “download” button in the top right corner like we did before, but this time choose “Download panel output”. A box will pop up asking you to choose which plots to download. This means you could have multiple plots being displayed in your panel but only choose to download a subset of them. Make sure “Feature assay plot” is checked and click “Download”. Your figure will be saved in a zip file in your Downloads folder.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/18J1x5rkWJisiv3EOMVZOma52XWxAVKANTg0Q2FXbOWU_g254b1a8b951_0_54.png" width="100%" /></p>
<div class="reflection">
<p><strong>Your turn!</strong></p>
<p>CD14 is a marker gene for the same type of cells as LYZ. Does it have the same cluster expression pattern as what we saw for LYZ?</p>
</div>
</div>
<div id="get-session-info" class="section level2 hasAnchor" number="11.6">
<h2><span class="header-section-number">11.6</span> Get Session Info<a href="single-cell-with-bioconductor-exercises.html#get-session-info" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It’s a good idea to document information about the packages (and their versions) you used while running the analysis. The last codeblock uses the <code>sessionInfo()</code> command to do just that. Here’s an example of what that might look like:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="single-cell-with-bioconductor-exercises.html#cb7-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.3.2 (2023-10-31)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 22.04.4 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## time zone: Etc/UTC
## tzcode source: system (glibc)
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## loaded via a namespace (and not attached):
##  [1] sass_0.4.8        utf8_1.2.4        generics_0.1.3    xml2_1.3.6       
##  [5] stringi_1.8.3     lattice_0.21-9    hms_1.1.3         digest_0.6.34    
##  [9] magrittr_2.0.3    evaluate_0.23     grid_4.3.2        timechange_0.3.0 
## [13] bookdown_0.41     fastmap_1.1.1     rprojroot_2.0.4   jsonlite_1.8.8   
## [17] Matrix_1.6-1.1    processx_3.8.3    chromote_0.3.1    ps_1.7.6         
## [21] promises_1.2.1    httr_1.4.7        fansi_1.0.6       ottrpal_1.3.0    
## [25] udpipe_0.8.11     cow_0.0.0.9000    jquerylib_0.1.4   cli_3.6.2        
## [29] rlang_1.1.4       gitcreds_0.1.2    cachem_1.0.8      yaml_2.3.8       
## [33] tools_4.3.2       tzdb_0.4.0        dplyr_1.1.4       curl_5.2.0       
## [37] vctrs_0.6.5       R6_2.5.1          lifecycle_1.0.4   lubridate_1.9.3  
## [41] snakecase_0.11.1  stringr_1.5.1     janitor_2.2.0     pkgconfig_2.0.3  
## [45] later_1.3.2       pillar_1.9.0      bslib_0.6.1       data.table_1.15.0
## [49] glue_1.7.0        Rcpp_1.0.12       highr_0.11        xfun_0.48        
## [53] tibble_3.2.1      tidyselect_1.2.0  knitr_1.48        textrank_0.3.1   
## [57] websocket_1.4.2   htmltools_0.5.7   igraph_2.0.2      rmarkdown_2.25   
## [61] webshot2_0.1.1    readr_2.1.5       compiler_4.3.2    askpass_1.2.0    
## [65] openssl_2.1.1</code></pre>
</div>
<div id="shutting-down" class="section level2 hasAnchor" number="11.7">
<h2><span class="header-section-number">11.7</span> Shutting Down<a href="single-cell-with-bioconductor-exercises.html#shutting-down" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="borrowed_chunk">
<ol style="list-style-type: decimal">
<li><p>Pausing your cloud environment only temporarily stops your work. When you are ready to delete the cloud environment, click on the RStudio icon on the right-hand side and select “Settings”. If you don’t see this icon, you may need to scroll to the right.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/1eypYLLqD11-NwHLs4adGpcuSB07dYEJfAaALSMvgzqw_ge1182913a6_0_41.png" alt="Screenshot of the Workspace page. The RStudio icon associated with the cloud environment is highlighted. The Settings button is also highlighted" width="100%" /></p></li>
<li><p>Click on “Delete Environment”.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/1eypYLLqD11-NwHLs4adGpcuSB07dYEJfAaALSMvgzqw_ge1182913a6_0_20.png" alt="Screenshot of the cloud environment popout. &quot;Delete environment&quot; is highlighted." width="100%" /></p></li>
<li><p>If you are certain that you do not need the data and configuration on your disk, you should select “Delete everything, including persistent disk”. If there is anything you would like to save, open the compute environment and copy the file(s) from your compute environment to another location, such as the Workspace bucket, GitHub, or your local machine, depending on your needs.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/1eypYLLqD11-NwHLs4adGpcuSB07dYEJfAaALSMvgzqw_ge1182913a6_0_46.png" alt="Screenshot of the cloud environment popout. &quot;Delete everything, including persistent disk&quot; is highlighted." width="100%" /></p></li>
<li><p>Select “DELETE”.</p>
<p><img src="03-single-cell-with-bioconductor_files/figure-html/1eypYLLqD11-NwHLs4adGpcuSB07dYEJfAaALSMvgzqw_ge1182913a6_0_51.png" alt="Screenshot of the cloud environment popout. &quot;Delete&quot; is highlighted." width="100%" /></p></li>
</ol>
<div class="notice">
<p>You can also delete your cloud environment(s) and disk storage at <a href="https://anvil.terra.bio/#clusters" class="uri">https://anvil.terra.bio/#clusters</a>.</p>
</div>
</div>
</div>
</div>
<hr>
<center> 
  <div class="footer">
      All illustrations <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY. </a>
      <br>
      All other materials <a href= "https://creativecommons.org/licenses/by/4.0/"> CC-BY </a> unless noted otherwise.
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="single-cell-with-bioconductor-preparation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="instructor-guide.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
